src/core/exec-invoke.c

```diff
@@ -2620,6 +2683,8 @@ static int setup_smack(
 static int compile_bind_mounts(
                 const ExecContext *context,
                 const ExecParameters *params,
+                uid_t uid,
+                gid_t gid,
```

Comment:
these are very generically named. I think they should make clear what they are applied to. Maybe exec_directory_uid/exec_directory_gid. Maybe also a comment.

After all, they are not applied the bind mounts created explicitly.